app/models/canonical_armor.rb

```diff
@@ -1,11 +1,25 @@
 # frozen_string_literal: true
 
 class CanonicalArmor < ApplicationRecord
-  has_many :enchantments, through: :canonical_armors_enchantments
-  has_many :smithing_materials, through: :canonical_armors_smithing_materials, source: :canonical_materials
-  has_many :tempering_materials, through: :canonical_armors_tempering_materials, source: :canonical_materials
+  has_many :canonical_armors_enchantments, dependent: :destroy
```

Comment:
This stuff was just wrong in the original PRs - I forgot you had to include this association in the model file.